node/add-to-path.js

```diff
@@ -0,0 +1,73 @@
+// add-to-path.js
+// This is a Node-side script to add Oni2 to their path.
+// This is done in node, to get sudo-prompt.
+const fs = require("fs");
+const os = require("os");
+const path = require("path");
+const sudo = require("sudo-prompt");
+
+const isWindows = () => os.platform() === "win32"
+const isMac = () => os.platform() === "darwin"
+const isLinux = () => os.platform() === "linux"
+
+const getLinkPath = () => (isMac() || isLinux() ? "/usr/local/bin/oni2" : "")
+
+const isAddedToPath = () => {
+    try {
+        fs.lstatSync(getLinkPath());
+    } catch (_) {
+        return false
+    }
+    return true
+}
+const removeFromPath = () => {
+    if (isAddedToPath() && !isWindows()) {
+        fs.unlinkSync(getLinkPath())
+    }
+}
+
+const addToPath = async () => {
+    if (!isAddedToPath() && !isWindows()) {
+        const appDirectory = path.join(path.dirname(process.mainModule.filename), "..");
+        let imgPath = path.join(appDirectory, "Onivim2.icns");
+
+        // TODO: Check this is valid for all use cases.
+        if (!fs.existsSync(imgPath)) {
+            imgPath = path.join(appDirectory, "assets", "images", "Onivim2.icns");
+        }
+
+        const options = { name: "Oni2", icns: imgPath };
+        let linkDest = "";
+
+        if (isMac()) {
+            // TODO: This isn't valid for dev builds, does that matter?
```

Comment:
This always links to a file over in the installed mac folder. Should it also work for a development build? I figured it wouldn't really be possible but perhaps I don't get the `esy` folder structure enough.

If not, it may make sense to gate the action and only make it accessible outside of dev builds. We need a gate in general to restrict this to mac/linux, so it doesn't appear in the menu on Windows.

Reply:
> This always links to a file over in the installed mac folder. Should it also work for a development build? I figured it wouldn't really be possible but perhaps I don't get the esy folder structure enough.

I think we'd want this to be relative to the executable - the `add-to-path.js` script should be in the same `Resources` folder as `run.sh`, so I believe we could just use `path.absolute(path.join(__dirname, "run.sh"))` here. This would't solve the dev build issue, but would help if the user was running it from another place (ie, if I run `esy create-release` -> run it from `open _release/Onivim2.app`.

> If not, it may make sense to gate the action and only make it accessible outside of dev builds. We need a gate in general to restrict this to mac/linux, so it doesn't appear in the menu on Windows.

Sounds good! For this PR - let's just gate this action for Mac-only.